### PR TITLE
Call fn_seek to filter out of range keys when fn_kind shows a snapshot of format tablet

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: ref #7256

Problem Summary:

In a tablet snapshot, out-of-range kvs may exist, we must filter them.

1. Introduce fn_seek and fn_kind in ProxyFFI.h.
2. If fn_kind == KIND_TABLET, we will seek to region's start_key, to avoid read out-of-range kvs, if start_key is not empty.
3. Meanwhile, if fn_kind == KIND_TABLET, we will stop read if the next key if greater or equal than end_key, if end_key is not empty.

The serverless proxy need to implement a dummy interface to adapt newer tiflash.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by lidezhu — APPROVED** [review]
LGTM

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Storages/DeltaMerge/SSTFilesToBlockInputStream.cpp [hunk 1]
```diff
@@ -228,6 +228,7 @@ void SSTFilesToBlockInputStream::loadCFDataFromSST(ColumnFamilyType cf, const De
                 BaseBuffView key = reader->keyView();
                 BaseBuffView value = reader->valueView();
                 // TODO: use doInsert to avoid locking
+                LOG_DEBUG(log, "! LLLLK3 {}", Redact::keyToDebugString(key.data, key.len));
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/SSTFilesToBlockInputStream.cpp:231]: this logging should be remove

File: dbms/src/Debug/MockRaftStoreProxy.cpp [hunk 2]
```diff
@@ -546,6 +551,8 @@ void MockRaftStoreProxy::doApply(
     if (cmd.has_raw_write_request())
     {
         // TiFlash write
+        std::string s;
+        google::protobuf::TextFormat::PrintToString(request, &s);
```
Comment (JaySon-Huang) [inline dbms/src/Debug/MockRaftStoreProxy.cpp:555]: remove debug logging?

**Pull request merged**